app/src/main/java/org/literacyapp/authentication/TrainingHelper.java

```diff
@@ -152,7 +170,7 @@ private boolean isStudentImageValid(StudentImage studentImage){
     }
 
     // Delete StudentImageCollectionEvent and all StudentImages if a file doesn't exist anymore or the feature extraction failed for another reason
-    private void deleteStudentImagesRecursive(StudentImage studentImage, String reason){
+    private synchronized void deleteStudentImagesRecursive(StudentImage studentImage, String reason){
```

Comment:
@jogrimst Can we move these changes to a separate pull request, as I think this would be too much for this one?
- introduce invalid / expired flag on StudentImage
- Move files to a different folder